### PR TITLE
chore: relocate bigtable-client-core configurations to bigtable-clien…

### PR DESCRIPTION
**Pull request by mutianf**
…t-core-config

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationAwaitCompletion.java [hunk 1]
```diff
@@ -104,7 +104,7 @@ public void setup() {
         Collections.synchronizedList(new ArrayList<ListenableFuture<MutateRowResponse>>());
 
     // Keep track of methods of completing mutateRowsAsync calls.  This method will add a Runnable
-    // that can set the correct value of the MutateRowsResponse future.
+    // that can set the correct value of the MutateRowsResponse future.O
```
Comment (diegomez17) [inline bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationAwaitCompletion.java:107]: nit: remove the zero/file from PR

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestRpcRetryBehavior.java [hunk 2]
```diff
@@ -31,7 +31,6 @@
 import com.google.bigtable.v2.ReadModifyWriteRowResponse;
 import com.google.bigtable.v2.ReadRowsRequest;
 import com.google.bigtable.v2.ReadRowsResponse;
-import com.google.cloud.bigtable.config.Logger;
```
Comment (diegomez17) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestRpcRetryBehavior.java:34]: Where is this file getting the Logger from w/o this import statement? Maybe the Logger was suppose to be removed like in FutureUtils.java?

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java [hunk 3]
```diff
@@ -292,7 +292,7 @@ public void testHBaseSnapshotImport() throws Exception {
 
     // Validate the counters.
     Map<String, Long> counters = getCountMap(result);
-    Assert.assertEquals(counters.get("ranges_matched"), (Long) 101L);
+    Assert.assertEquals(counters.get("ranges_matched"), (Long) 100L);
```
Comment (diegomez17) [inline bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java:295]: Where is the logic for this being changed? I don't see anything that should affect this counter in this PR.

Comment (mutianf) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java:295]: This was changed in https://github.com/googleapis/java-bigtable-hbase/pull/3586

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java [hunk 4]
```diff
@@ -15,6 +15,10 @@
  */
 package com.google.cloud.bigtable.config;
 
+import static com.google.cloud.bigtable.config.CredentialOptions.JsonCredentialsOptions;
+import static com.google.cloud.bigtable.config.CredentialOptions.P12CredentialOptions;
+import static com.google.cloud.bigtable.config.CredentialOptions.UserSuppliedCredentialOptions;
+
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java:21]: why are these static now?

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml [hunk 5]
```diff
@@ -331,11 +331,15 @@ limitations under the License.
...
+    <dependency>
+      <groupId>com.google.cloud.bigtable</groupId>
+      <artifactId>bigtable-client-core-config</artifactId>
+      <version>${project.version}</version>
+    </dependency>
     <dependency>
       <groupId>com.google.cloud.bigtable</groupId>
       <artifactId>bigtable-hbase</artifactId>
-      <version>1.27.2-SNAPSHOT</version> <!-- {x-version-update:bigtable-client-parent:current} -->
+      <version>${project.version}</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml:342]: need update tag?

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml [hunk 6]
```diff
@@ -363,7 +367,7 @@ limitations under the License.
     <dependency>
       <groupId>com.google.cloud.bigtable</groupId>
       <artifactId>bigtable-hbase-integration-tests-common</artifactId>
-      <version>1.27.2-SNAPSHOT</version> <!-- {x-version-update:bigtable-client-parent:current} -->
+      <version>${project.version}</version> <!-- {x-version-update:bigtable-client-parent:current} -->
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml:370]: lets keep this as is

**Review by igorbernstein2 — APPROVED** [review]

**Pull request merged**